src/build123d/geometry.py

```diff
@@ -1145,6 +1145,13 @@ def __mul__(self, other: Location) -> Location:
     def __pow__(self, exponent: int) -> Location:
         return Location(self.wrapped.Powered(exponent))
 
+    def __eq__(self, other: Location):
+        return (
+            isinstance(other, Location)
+            and self.position == other.position
+            and self.orientation == other.orientation
```

Comment:
This doesn't cover all the possibilities since the orientations are angles so mod 360 applies but more significantly different combinations result in the same end angle.  It seems safer to compare quaternion's 
```py
        quaternion1 = gp_Quaternion()
        quaternion.SetEulerAngles(
            gp_EulerSequence.gp_Intrinsic_XYZ,
            radians(x1),
            radians(y1),
            radians(z1),
        )
        quaternion2 = ...
        ... quaternion1.IsEqual(quaternion2)
```

```